spawn_worker/src/unix/spawn.rs

```diff
@@ -183,71 +319,138 @@ impl SpawnCfg {
             }
         }
 
-        type SpawnFn = dyn Fn(&ExecVec, &ExecVec);
+        // setup arbitrary fd passing
+        let _shorter_lived_fd = if let Some(src_fd) = &self.fd_to_pass {
+            // we're stripping the close on exec flag from the FD
+            // to ensure we will not modify original fd, whose expected lifetime is unknown
+            // we should clone the FD that needs passing to the subprocess, keeping its lifetime
+            // as short as possible
```

Comment:
I disagree with this, on fork all of the processes fd gets cloned, so I am not sure what cloning an fd to a "smaller" lifetime improves..
On the `CLOEXEC` argument I am split.
* Passing a fd with the flag set should be an error since the creator of the fd specified that they don't want them to get passed to forks
* On the other hand, by default `try_clone` on io-lifetimes fds adds this flag https://doc.rust-lang.org/src/std/os/fd/owned.rs.html#94 . But IMO the fix should be either stopping to clone OwnedFds all over the place and use Borrowed ones instead, or use our own clone function that skips this flag.